docs/wiki/[Examples]-Create-and-Assign-Custom-RBAC-Roles.md

```diff
@@ -0,0 +1,128 @@
+
+### `lib/role_definitions/role_definition_es_reader_support_tickets.tmpl.json`
+
+```json
+{
+    "name": "8753b9c9-07db-4073-98fb-4a7ab1e2ffe5",
+    "type": "Microsoft.Authorization/roleDefinitions",
+    "apiVersion": "2018-01-01-preview",
+    "properties": {
+        "roleName": "Reader-Support-Tickets",
+        "description": "View everything in the subscription and also open support tickets.",
+        "type": "customRole",
+        "permissions": [
+            {
+                "actions": [
+                    "*/read",
+                    "Microsoft.Support/*"
+                ],
+                "notActions": [],
+                "dataActions": [],
+                "notDataActions": []
+            }
+        ],
+        "assignableScopes": [
+            "${current_scope_resource_id}"
+        ]
+    }
+}
+```
+
+## Assign Role
+
+To assign a custom role, we need to expand upon the built-in configuration by using `archetype extensions`.
+You can learn more about archetype extensions in [this article](https://github.com/Azure/terraform-azurerm-caf-enterprise-scale/wiki/%5BExamples%5D-Expand-Built-in-Archetype-Definitions).
+
+If you don't already have an `archetype_extension_es_root.tmpl.json` file within your custom `lib` directory, create one and copy the below code in to the file.
+This code will save your new `Reader-Support-Tickets` role at this scope and allow it to be used either at the `es_root` scope or below.
+
+### `lib/archetype_extension_es_root.tmpl.json`
+
+```json
+{
+  "extend_es_root": {
+    "policy_assignments": [],
+    "policy_definitions": [],
+    "policy_set_definitions": [],
+    "role_definitions": ["Reader-Support-Tickets"],
+    "archetype_config": {
+      "parameters": {},
+      "access_control": {}
+    }
+  }
+}
+```
+
+If you don't already have an `archetype_extension_es_landing_zones.tmpl.json` file within your custom `lib` directory, create one and copy the below code in to the file.
+This code will assign your new `Reader-Support-Tickets` role to a group named `Contoso Reader and Support Tickets`.
+In order to assign the `Reader-Support-Tickets` role to the group, you need to use the groups objectID which can be located in Azure Active Directory.
+
+>IMPORTANT: Due to how the module works, you must prefix your group name with the location at which it has been saved. In our example this would be `"[CONTOSO]"`.
+If we had saved our role at the `es_landing_zones` scope then we would use a prefix of `"[CONTOSO-LANDING-ZONES]"`
+
+### `lib/archetype_extension_es_landing_zones.tmpl.json`
+
+```json
+{
+  "extend_es_landing_zones": {
+    "policy_assignments": [],
+    "policy_definitions": [],
+    "policy_set_definitions": [],
+    "role_definitions": [],
+    "archetype_config": {
+      "parameters": {},
+      "access_control": {
+        "[CONTOSO] Reader-Support-Tickets": ["2f449a58-06b8-48a2-b60c-9107be0ca878"]
```

Comment:
Is it worth making it a little clearer that this GUID is the object ID for the group (as mentioned above)?

Reply:
Thanks Kevin, I've added a note to remind readers to update the group ObjectID.